test/unit/kubernetes-deploy/concurrency_test.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+class ConcurrencyTest < KubernetesDeploy::TestCase
+  class TestWork
+    attr_accessor :worked_by_threads
+    def initialize
+      @worked_by_threads = []
+    end
+  end
+
+  def test_split_across_threads_raises_without_a_block
+    assert_raises_message(ArgumentError, "Block of work is required") do
+      KubernetesDeploy::Concurrency.split_across_threads([TestWork.new])
+    end
+  end
+
+  def test_split_across_threads_works_with_zero_work
+    KubernetesDeploy::Concurrency.split_across_threads([]) do |individual|
+      individual.worked_by_threads << Thread.current.object_id
+    end
+    assert_work_distribution([], [])
```

Comment:
`assert_nothing_raised` is just `yield` and some developers try to avoid it. Without `assert_nothing_raised ` if something raises, the test fails anyway.